superbench/benchmarks/micro_benchmarks/cudnn_test/cudnn_benchmark.h

```diff
@@ -0,0 +1,254 @@
+// Copyright(c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file cudnn_benchmark.h
+ * @brief Unify the base class for cudnn function benchmark
+ */
+
+#pragma once
+
+#include <chrono>
+#include <iomanip>
+#include <tuple>
+#include <unordered_map>
+
+#include "cudnn_helper.h"
+
+/**
+ * @brief Enum of cudnn function name
+ */
+namespace cudnn_test {
+enum cudnn_function_name_enum {
+    e_cudnnConvolutionForward = 0,
+    e_cudnnConvolutionBackwardData,
+    e_cudnnConvolutionBackwardFilter,
+};
+
+/**
+ * @brief Map from cudnn function name to cudnn function name enum
+ */
+static std::unordered_map<std::string, cudnn_function_name_enum> const cudnn_function_name_string = {
+    {"cudnnConvolutionForward", cudnn_function_name_enum::e_cudnnConvolutionForward},
+    {"cudnnConvolutionBackwardData", cudnn_function_name_enum::e_cudnnConvolutionBackwardData},
+    {"cudnnConvolutionBackwardFilter", cudnn_function_name_enum::e_cudnnConvolutionBackwardFilter},
+
+};
+
+/**
+ * @brief Class to store the configuration of cudnn function params
+ */
+class CudnnConfig {
+  protected:
+    int num_test;
+    int warm_up;
+    int num_in_step;
+    int random_seed;
+    std::string name;
+    cudnn_function_name_enum e_name;
+    std::vector<int> input_dims_;
+    std::vector<int> input_stride_;
+    std::vector<int> filter_dims_;
+    std::vector<int> output_dims_;
+    std::vector<int> output_stride_;
+    int algo_;
+    int arrayLength_;
+    std::vector<int> padA_;
+    std::vector<int> filterStrideA_;
+    std::vector<int> dilationA_;
+    cudnnConvolutionMode_t mode_;
+    bool use_tensor_core_;
+    cudnnDataType_t input_type_;
+    cudnnDataType_t conv_type_;
+    std::string function_str_;
+
+  public:
+    void set_num_test(int num_test) { this->num_test = num_test; }
+    void set_warm_up(int warm_up) { this->warm_up = warm_up; }
+    void set_num_in_step(int num_in_step) { this->num_in_step = num_in_step; }
+    void set_random_seed(int random_seed) { this->random_seed = random_seed; }
+    void set_name(std::string &n) { name = n; }
```

Comment:
For all reference type, use 'const' type? Like:
`const std::string &n`

Reply:
ok, thank you for the comment